src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs

```diff
@@ -16,7 +33,158 @@ public static class MvcServiceCollectionExtensions
         public static IServiceCollection AddMvc([NotNull] this IServiceCollection services)
         {
             ConfigureDefaultServices(services);
-            services.TryAdd(MvcServices.GetDefaultServices());
+
+            // Options and core services.
+            services.TryAdd(ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, MvcOptionsSetup>());
+            services.TryAdd(
```

Comment:
This two probably should be a normal Adds, as multiple configure options are allowed and are run in order (its an IEnueramble